frontend/src/components/Table/HorizontalTable.tsx

```diff
@@ -23,29 +32,37 @@ export function HorizontalTable({
   expandElement,
   optionElement,
 }: HorizontalTableProps) {
-  const [page, setPage] = React.useState(0);
-  const [rowsPerPage, setRowsPerPage] = React.useState(10);
-  const [sortBy, setSortBy] = React.useState("");
-  const [sortOrder, setSortOrder] = React.useState("desc");
-  const [expandedRow, setExpandedRow] = React.useState<number | null>(null);
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [sortBy, setSortBy] = useState("");
+  const [sortOrder, setSortOrder] = useState("desc");
+  const [expandedRow, setExpandedRow] = useState<number | null>(null);
   const [sortedContent, setSortedContent] =
-    React.useState<(string | number | React.ReactNode)[][]>(content);
+    useState<(string | number | ReactNode)[][]>(content);
 
-  const sortContent = React.useCallback(
+  const sortContent = useCallback(
     (
-      content: (string | number | React.ReactNode)[][],
+      content: (string | number | ReactNode)[][],
       column: string,
       order: string,
-    ): (string | number | React.ReactNode)[][] => {
-      return content.slice().sort((a, b) => {
-        const comparison =
-          a[header.indexOf(column)]! > b[header.indexOf(column)]! ? 1 : -1;
+    ): (string | number | ReactNode)[][] => {
+      return [...content].sort((a, b) => {
+        let comparison = 0;
+        const columnIndex = header.indexOf(column);
+        if (isValidElement(a[columnIndex])) {
+          const aToCompare = a[columnIndex] as ReactElement;
+          const bToCompare = b[columnIndex] as ReactElement;
+          comparison =
+            aToCompare.props.children > bToCompare.props.children ? 1 : -1;
```

Comment:
This will break if the content is wrapped around 2 component or more i.e.` <NavLink> <Button> 123 </Button> </NavLink>` which is our current implementation for the links in the master branch. Maybe we could mark some of the column to have its data points wrapped in some function that we can parse into the Table component so that the Table could have access to the raw data and properly render it using the wrapper function that return the desired layout like NavLink -> Button -> Text.

Reply:
@Anh-Duy-Tran Is it just me or this doesn't break? I have merged `master` and test and it works fine

Reply:
@ducanhpham0312  this is my fresh build of the app from this branch, and the column with the Navlink and Button is still messed up
<img width="303" alt="Screenshot 2023-12-06 at 4 45 14" src="https://github.com/ducanhpham0312/zeevision/assets/113171462/06867dcf-2d4e-4d59-bcb5-5259fd86434f">
<img width="295" alt="Screenshot 2023-12-06 at 4 46 56" src="https://github.com/ducanhpham0312/zeevision/assets/113171462/fc3f5e1f-3c48-4389-b9e2-769bab37494b">